apollo-utilities dependency causing peer dependency warnings on graphql@16+

When upgrading dependencies to latest, npm complains that the `apollo-utilities` dependency [wants graphql@15](https://github.com/apollographql/federation/blob/e8aab43c41f847a7ce2511034669448cd4be8706/gateway-js/package.json#L40), while federation wants [graphql@16](https://github.com/apollographql/federation/blob/e8aab43c41f847a7ce2511034669448cd4be8706/gateway-js/package.json#L48).

From what I can see via [code search](https://github.com/apollographql/federation/search?q=apollo-utilities), the only usage of apollo-utilities [is in the test suite](https://github.com/apollographql/federation/blob/1ffecef9ef52f6dc127939cdc375f0465774f093/gateway-js/src/__tests__/execution-utils.ts#L17); the peer dependency issue with graphql aside, it would seem that this library belongs in devDependencies?


```bash
⇒  npm i
npm WARN ERESOLVE overriding peer dependency
npm WARN While resolving: apollo-utilities@1.3.4
npm WARN Found: graphql@16.3.0
npm WARN node_modules/graphql
npm WARN   graphql@"16.3.0" from the root project
npm WARN   18 more (@apollo/core-schema, @apollo/federation, ...)
npm WARN 
npm WARN Could not resolve dependency:
npm WARN peer graphql@"^0.11.0 || ^0.12.0 || ^0.13.0 || ^14.0.0 || ^15.0.0" from apollo-utilities@1.3.4
npm WARN node_modules/@apollo/gateway/node_modules/apollo-utilities
npm WARN   apollo-utilities@"^1.3.0" from @apollo/gateway@0.50.0
npm WARN   node_modules/@apollo/gateway
npm WARN 
npm WARN Conflicting peer dependency: graphql@15.8.0
npm WARN node_modules/graphql
npm WARN   peer graphql@"^0.11.0 || ^0.12.0 || ^0.13.0 || ^14.0.0 || ^15.0.0" from apollo-utilities@1.3.4
npm WARN   node_modules/@apollo/gateway/node_modules/apollo-utilities
npm WARN     apollo-utilities@"^1.3.0" from @apollo/gateway@0.50.0
npm WARN     node_modules/@apollo/gateway
```

My package.json:
```json
{
  "name": "graphql-federation-gateway-lambda",
  "version": "1.0.0",
  "dependencies": {
    "@apollo/gateway": "0.50.0",
    "@newrelic/apollo-server-plugin": "1.2.3",
    "apollo-server-core": "3.6.7",
    "apollo-server-lambda": "3.6.7",
    "aws-sdk": "2.1107.0",
    "aws-xray-sdk-core": "3.3.4",
    "bunyan": "1.8.15",
    "graphql": "16.3.0",
    "graphql-depth-limit": "1.1.0",
    "newrelic": "8.9.1"
  },
  "devDependencies": {
    "@shutterstock/eslint-config": "2.1.1",
    "chai": "4.3.6",
    "eslint": "8.12.0",
    "eslint-plugin-import": "2.25.4",
    "mocha": "9.2.2",
    "nock": "13.2.4",
    "sinon": "13.0.1",
    "supertest": "6.2.2"
  }
}
```

Thanks!

--- a/gateway-js/package.json
+++ b/gateway-js/package.json
@@ -37,7 +37,6 @@
     "apollo-server-env": "^3.0.0 || ^4.0.0",
     "apollo-server-errors": "^2.5.0 || ^3.0.0",
     "apollo-server-types": "^0.9.0 || ^3.0.0",
-    "apollo-utilities": "^1.3.0",
     "async-retry": "^1.3.3",
     "loglevel": "^1.6.1",
     "make-fetch-happen": "^8.0.0",

--- a/gateway-js/src/__tests__/execution-utils.ts
+++ b/gateway-js/src/__tests__/execution-utils.ts
@@ -11,7 +11,7 @@ import {
 } from '@apollo/gateway';
 import { QueryPlan, QueryPlanner } from '@apollo/query-planner';
 import { LocalGraphQLDataSource } from '../datasources/LocalGraphQLDataSource';
-import { mergeDeep } from 'apollo-utilities';
+import { mergeDeep } from '@apollo/client/utilities';
 
 import { queryPlanSerializer, astSerializer } from 'apollo-federation-integration-testsuite';
 import gql from 'graphql-tag';

